### ISSUE TITLE
Mage Water Elemental: NPCs remain aggro'd on Elemental after Elemental's death.

### ISSUE THREAD

**Issue by Zelwindin** [dalaranwow/dalaran-wow#3749]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (OLD - ... tags should no longer be used)
[//]: # (Priority - ... tags are assigned by team members)

**Description**: NPCs remain aggro'd on Water Elemental after Elemental's death.

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)

So, basically what happens here is that when you are fighting an NPC (generally an elite NPC) and you frost nova the NPC right next to the Water Elemental, the NPC will attack the Elemental with melee attacks (this works just like retail). HOWEVER, once the NPC kills the elemental, the NPC will just stand there, still aggro'd on the corpse of the Elemental.

Only once the Elemental's corpse despawns does the NPC drop aggro on the Elemental and go back to attacking the player.

NOTE: I am using the glyph that makes the Water Elemental permanent. This bug may not occur without that glyph.

**Expected** behaviour**: Tell us what should happen instead**

A) NPC should attack the object with the highest threat on it as soon as Frost Nova ends.
B) NPC should attack a new target with the highest threat once it kills the Water Elemental.

**Steps to reproduce the problem**:

1. Spec deep frost on a Mage, pick up water elemental.
2. Equip the glyph that makes water elemental permanent.
3. Engage an elite NPC that takes awhile to kill.
4. Kite the NPC to be right next to the Water Elemental.
5. Frost Nova the NPC in place, adjacent to the Elemental.
6. NPC will engage Elemental in melee.
7. NPC kills Elemental.
8. NPC remains aggro'd on the Elemental's corpse.
9. Easily solo any NPC by yourself, because it will not aggro on you and will stand there and die.

**Include proofs for this behaviour**

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

LMK if you need proof, I can try and take some screenshots. It's easily testable though.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

Summon Water Elemental ID 31687
Frost Nova ID 122

You can research this on several pages like http://wotlk.openwow.com or http://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by Golganeth** [comment]
#828
